Components/Sensors/Inc/IMUTask.hpp

```diff
@@ -0,0 +1,57 @@
+/**
+ ******************************************************************************
+ * File Name          : IMUTask.hpp
+ * Description        :
+ ******************************************************************************
+*/
+#ifndef SOAR_SENSOR_IMU_TASK_HPP_
+#define SOAR_SENSOR_IMU_TASK_HPP_
+/* Includes ------------------------------------------------------------------*/
+#include "Task.hpp"
+#include "Data.h"
+#include "SystemDefines.hpp"
+
+
+/* Macros/Enums ------------------------------------------------------------*/
+enum IMU_TASK_COMMANDS {
+	IMU_NONE = 0,
+	IMU_REQUEST_NEW_SAMPLE,// Get a new IMU sample, task will be blocked for polling time
+	IMU_REQUEST_TRANSMIT,	// Send the current IMU data over the Radio
+	IMU_REQUEST_DEBUG		// Send the current IMU data over the Debug UART
+};
+
+/* Class ------------------------------------------------------------------*/
+class IMUTask : public Task
+{
+public:
+	static IMUTask& Inst() {
+		static IMUTask inst;
+		return inst;
+	}
+
+	void InitTask();
+
+protected:
+	static void RunTask(void* pvParams) { IMUTask::Inst().Run(pvParams); } // Static Task Interface, passes control to the instance Run();
+
+	void Run(void* pvParams);	// Main run code
+	
+	void HandleCommand(Command& cm);
+	void HandleRequestCommand(uint16_t taskCommand);
+
```

Comment:
add void SampleMeanBias(); for sampling the power-on bias as a stationary calibration step. The procedure will look like this:

1. increase the output data rate
2. continuously read the IMU for ~5 samples
3. calculate average of accel and gyro readings in all three axes, ignoring the first ~100 samples (the IMU data is usually not stable at the start)
5. update a power on bias struct using averages, accounting for gravity in accel_z
6. reset output data rate

We could then apply this offset on each sample read to "normalize" them (ie. remove the random electrical noise)